### PR TITLE
[s] Makes it so non-admins cannot activate the Ark

### PR DESCRIPTION
**Pull request by ghost**
LITERALLY ANY BORED GHOST CAN CLICK THE ARK

:cl: More Robust Than You
admin: ONLY ADMINS CAN ACTIVATE THE ARK NOW
/:cl:

### CODE REVIEW

**Comment by nicbn** [comment]
wait





what

**Comment by DoctoreMad** [comment]
ALL CHECKS HAVE FAILED
1 FAILING CHECK

**Comment by praisenarsie** [comment]
its a rippie issue, not a tg issue guys lol

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 1]
```diff
@@ -266,7 +266,7 @@
 					CHECK_TICK
 
 /obj/structure/destructible/clockwork/massive/celestial_gateway/attack_ghost(mob/user)
-	if(!user.client || !user.client.holder)
+	if(!user.client || !user.client.holder || (user.client && !user.client.check_rights(R_ADMIN)))
```
Comment (lzimann) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: the holder check should already prevent anything from happening

Comment (MoreRobustThanYou) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: _cough_ hippie mentors _cough_

Comment (praisenarsie) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: its a rippie issue, not a tg issue guys lol

**Comment by optimumtact** [comment]
code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm:269:error: user.client.check_rights: undefined proc

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm [hunk 2]
```diff
@@ -266,7 +266,7 @@
 					CHECK_TICK
 
 /obj/structure/destructible/clockwork/massive/celestial_gateway/attack_ghost(mob/user)
-	if(!user.client || !user.client.holder)
+	if(!user.client || !user.client.holder || !IsAdminGhost(user))
```
Comment (JJRcop) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: ```user.client.holder``` does this though, right?

Comment (praisenarsie) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: ITS A FUCKING RIPPIE ISSUE, NOT A /TG/ ISSUE

Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: `!user.client.holder` does this

**Review by ChangelingRain — DISMISSED** [review]

**Comment by DoctoreMad** [comment]
It can't be a hippie issue since hippie is dead

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm (hunk 2 above)
Comment (praisenarsie) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: ITS A FUCKING RIPPIE ISSUE, NOT A /TG/ ISSUE

Comment (vuonojenmustaturska) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: just fucking replace the whole thing with IsAdminGhost(user), see https://github.com/tgstation/tgstation/blob/626302c1bf4bb2a37aa6508145751f6c4c78696d/code/modules/mob/mob_helpers.dm#L432-L443

Comment (vuonojenmustaturska) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: check_rights_for is defined in https://github.com/tgstation/tgstation/blob/d3dcc118ecb5f5410de684b1bb46289f8066ab23/code/modules/admin/holder2.dm#L130-L136

**Comment by jughu** [comment]
Didnt spacehippie step back in to revive hippie

**Review by Cyberboss — DISMISSED** [review]

File: code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (lzimann) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: Isadminghost is better

[the remark this answers is not in the document]
Comment (Cyberboss) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: isadminghost doesnt work for livingmins though

[the remark this answers is not in the document]
Comment (lzimann) [reply code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm]: good thing this is on `attack_ghost` then

**Comment by jughu** [comment]
THISS IS A HIPPIE ISSUE

**Pull request merged**